### PR TITLE
SLLS-204 Open issue in IDE - allow users to continue the flow without jumping back to the browser

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarsource/sonarlint/ls/clientapi/SonarLintVSCodeClient.java [hunk 1]
```diff
@@ -135,17 +156,64 @@ public void showIssue(ShowIssueParams showIssueParams) {
...
+    return currentConnections;
   }
 
   @Override
-  public CompletableFuture<org.sonarsource.sonarlint.core.clientapi.client.binding.AssistBindingResponse>
-  assistBinding(org.sonarsource.sonarlint.core.clientapi.client.binding.AssistBindingParams params) {
-    server.showHotspotOrIssueHandleNoBinding(params);
-    return CompletableFuture.failedFuture(new UnsupportedOperationException());
+  public CompletableFuture<AssistBindingResponse> assistBinding(org.sonarsource.sonarlint.core.clientapi.client.binding.AssistBindingParams params) {
+    return CompletableFutures.computeAsync(cancelChecker -> client.assistBinding(params)
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/clientapi/SonarLintVSCodeClient.java:192]: If I'm not mistaken, `client.assistBinding(params)` already returns a `CompletableFuture`, so I believe we can remove the `computeAsync` wrapper and the `join` below - less is more 😄

File: src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java [hunk 2]
```diff
@@ -183,6 +189,29 @@ private Optional<ProjectBindingWrapper> getBinding(Optional<WorkspaceFolderWrapp
     });
   }
 
+  public CompletableFuture<String> getUpdatedBindingForWorkspaceFolder(URI folderUri) {
+    var countDown = new CountDownLatch(1);
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java:193]: I would maybe rename this variable to `bindingUpdatedLatch` to make it more explicit what we are waiting for.

**Review by jblievremont — COMMENTED** [review]
Nice! See my comments - I'm not 100% sure about the simplification of the `CompletableFuture` handling, I believe it would need to be validated by manual test.

File: src/main/java/org/sonarsource/sonarlint/ls/clientapi/SonarLintVSCodeClient.java (hunk 1 above)
Comment (vojtech-suchy-sonarsource) [reply src/main/java/org/sonarsource/sonarlint/ls/clientapi/SonarLintVSCodeClient.java:192]: We had it that way originally but the futures were timing out. After consulting with Julien it turned out we had to wrap the future in an extra asynchronous computation. The reasoning was something along those lines:
_The lsp4j message processor is single-threaded so only one message is processed at a time. If the message is a request, the completable future will be added to a queue, waiting for completion. To not block other messages, we have to return a completable future immediately._

**Review by sophio-japharidze-sonarsource — APPROVED** [review]

**Pull request merged**